service-base/src/main/java/org/eclipse/hono/service/AbstractApplication.java

```diff
@@ -104,6 +105,17 @@ public final void setApplicationConfiguration(final ApplicationConfigProperties
         this.config = Objects.requireNonNull(config);
     }
 
+    /**
+     * Sets the HealthCheckServer for this application.
+     * 
+     * @param healthCheckServer The HealthCheckServer.
+     * @throws NullPointerException if healthCheckServer is {@code null}.
+     */
+    @Autowired
```

Comment:
I wonder if this could/should be marked required = false. If no health check server is registered, the service will still work properly, wouldn't it?

Reply:
Would be possible but require some minor adjustments to prevent NullPointers. Currently there is only one situation where you don't have health checks. That is when no port for the `HealthCheckServer` is configured. In that case it logs "No health check configured.". Making the bean not required would result two possible situations where you don't have health checks: if you don't provide the bean and if you do provide it but have no port configured. If we change this, I would suggest that we add an additional log statement that informs that no health check is provided.
I don't think one of the options is clearly better. What do you think?

Reply:
Before digging into the details, do we agree that making the health check server optional is a good idea?

Reply:
I am not so sure about making the health checks optional. If you explicitly doen't want one, then this may be a valid decision. If however, by accident, there is no health server present, but the setup assumes one to be present, then you would never know, as nothing fails.

So maybe having a noop-implementation of a health server would be a better approach.

Reply:
> I am not so sure about making the health checks optional. If you explicitly doen't want one, then this may be a valid decision. If however, by accident, there is no health server present, but the setup assumes one to be present, then you would never know, as nothing fails.

I agree.

> So maybe having a noop-implementation of a health server would be a better approach.

IMHO this is what we currently have.

Reply:
> So maybe having a noop-implementation of a health server would be a better approach.

@ctron Are you good with the current situation, where it is mandatory to have a  `HealthCheckServer` bean, but it can be disabled by just not setting a port for the health check (plus the log statement informing you that you don't have a health check configured)? Or do you want a noop-implementation of the `HealthCheckServer` bean?